homeassistant/components/statistics/sensor.py

```diff
@@ -358,15 +359,14 @@ def __init__(
         self.samples_keep_last: bool = samples_keep_last
         self._precision: int = precision
         self._percentile: int = percentile
-        self._value: StateType | datetime = None
```

Comment:
Just for my understanding. Why did you add this change? If I remember correctly StateType included float as well as int, which you have neglected now.

Reply:
It's not exactly related to this PR but why have a big generic typing when we know what it will return more specifically.

Reply:
Because value can be `int` :)

https://github.com/home-assistant/core/blob/2453e1284f479c79062f53588a36d71f0d9263c3/homeassistant/components/statistics/sensor.py#L720

Reply:
Good remark, didn't see that and mypy didn't complain. I will fix that